### PR TITLE
Remove vulnerability data and deprecate code

### PR DESCRIPTION
**Pull request by thc202**
Remove the vulnerability data (main and translations).
Deprecate vulnerability related code for removal.
Allow to set the provider of the vulnerability data for the Common Library add-on to provide that data to existing/old code.

Part of #8012.

---
WIP until the changes in Common Library add-on are merged.

### CODE REVIEW

**Comment by thc202** [comment]
This one can be reviewed even if WIP (it's just to avoid merging before the commonlib PR).

**Review by psiinon — APPROVED** [review]

File: zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java [hunk 1]
```diff
@@ -93,8 +54,16 @@ private static synchronized void init() {
...
+        var local = provider;
+        if (local == null) {
+            logNoProvider();
+            return List.of();
+        }
+        return local.getAll();
+    }
+
+    private static void logNoProvider() {
+        LOGGER.error("No provider found.", new Exception());
```
Comment (kingthorin) [inline zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java:66]: Should the new Exception have some details? If not why not just skip it? Should the literal string say "vulnerability data provider"?

Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java:66]: The exception is just to create a stack trace (for debug purposes). The logged class name already indicates it's for vulnerabilities.

File: zap/src/main/java/org/zaproxy/zap/extension/lang/LangImporter.java [hunk 2]
```diff
@@ -110,7 +110,7 @@ public static String importLanguagePack(String languagePack) {
 
     /**
      * Creates a {@code Pattern} to match filenames of, source and translated, resource files
-     * Messages.properties and vulnerabilities.xml.
+     * Messages.properties.
```
[the remark this answers is not in the document]
Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/extension/lang/LangImporter.java:113]: Was `filesnames` on purpose?

[the remark this answers is not in the document]
Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/extension/lang/LangImporter.java]: Tweaked without that one.

File: zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java (hunk 1 above)
Comment (kingthorin) [reply zap/src/main/java/org/zaproxy/zap/model/Vulnerabilities.java:66]: Ok

**Pull request merged**